pkgs/native_assets_cli/test/api/build_config_test.dart

```diff
@@ -0,0 +1,218 @@
+    expect(config1.linkModePreference, config2.linkModePreference);
+    expect(config1.dependencyMetadata, config2.dependencyMetadata);
+  });
+
+  test('BuildConfig fromConfig', () {
+    final buildConfig2 = BuildConfig(
+      outDir: outDirUri,
+      packageName: packageName,
+      packageRoot: packageRootUri,
+      targetArchitecture: Architecture.arm64,
+      targetOs: OS.android,
+      targetAndroidNdkApi: 30,
+      buildMode: BuildMode.release,
+      linkModePreference: LinkModePreference.preferStatic,
+    );
+
+    final config = Config(fileParsed: {
+      'build_mode': 'release',
+      'dry_run': false,
+      'link_mode_preference': 'prefer-static',
+      'out_dir': outDirUri.toFilePath(),
+      'package_name': packageName,
+      'package_root': packageRootUri.toFilePath(),
+      'target_android_ndk_api': 30,
+      'target_architecture': 'arm64',
+      'target_os': 'android',
+      'version': BuildOutput.version.toString(),
+    });
+
+    final fromConfig = BuildConfig.fromConfig(config);
+    expect(fromConfig, equals(buildConfig2));
+  });
+
+  test('BuildConfig.dryRun', () {
+    final buildConfig2 = BuildConfig.dryRun(
+      outDir: outDirUri,
+      packageName: packageName,
+      packageRoot: packageRootUri,
+      targetOs: OS.android,
+      linkModePreference: LinkModePreference.preferStatic,
+    );
+
+    final config = Config(fileParsed: {
+      'dry_run': true,
+      'link_mode_preference': 'prefer-static',
+      'out_dir': outDirUri.toFilePath(),
+      'package_name': packageName,
+      'package_root': packageRootUri.toFilePath(),
+      'target_os': 'android',
+      'version': BuildOutput.version.toString(),
+    });
+
+    final fromConfig = BuildConfig.fromConfig(config);
+    expect(fromConfig, equals(buildConfig2));
+  });
+
+  test('BuildConfig == dependency metadata', () {
+    final buildConfig1 = BuildConfig(
+      outDir: outDirUri,
+      packageName: packageName,
+      packageRoot: tempUri,
+      targetArchitecture: Architecture.arm64,
+      targetOs: OS.android,
+      targetAndroidNdkApi: 30,
+      buildMode: BuildMode.release,
+      linkModePreference: LinkModePreference.preferStatic,
+      dependencyMetadata: {
+        'bar': const Metadata({
+          'key': 'value',
+          'foo': ['asdf', 'fdsa'],
+        }),
+        'foo': const Metadata({
+          'key': 321,
+        }),
+      },
+    );
+
+    final buildConfig2 = BuildConfig(
+      outDir: outDirUri,
+      packageName: packageName,
+      packageRoot: tempUri,
+      targetArchitecture: Architecture.arm64,
+      targetOs: OS.android,
+      targetAndroidNdkApi: 30,
+      buildMode: BuildMode.release,
+      linkModePreference: LinkModePreference.preferStatic,
+      dependencyMetadata: {
+        'bar': const Metadata({
+          'key': 'value',
+        }),
+        'foo': const Metadata({
+          'key': 123,
+        }),
+      },
+    );
+
+    expect(buildConfig1, equals(buildConfig1));
```

Comment:
```suggestion
    expect(buildConfig1, equals(buildConfig2));
```

I assume? Otherwise I don't understand what is tested here.

Reply:
No, it's just testing that the equals method does _not_ say that to identical objects are inequal. Which can happen if you mess up the implementation.